webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.test.tsx

```diff
@@ -18,68 +21,70 @@ describe('components/AdminConsole/CustomEnableDisableGuestAccountsSetting', () =
         showConfirm: false,
     };
 
+    const intlProviderProps = {
+        defaultLocale: 'en',
+        locale: 'en',
+    };
+
+    const renderComponent = (props = {}) => {
+        return render(
+            <IntlProvider {...intlProviderProps}>
+                <Provider store={store}>
+                    <CustomEnableDisableGuestAccountsSetting
+                        {...baseProps}
+                        {...props}
+                    />
+                </Provider>
+            </IntlProvider>);
+    };
+
     const warningMessage = (
         <FormattedMessage
             defaultMessage='All current guest account sessions will be revoked, and marked as inactive'
             id='admin.guest_access.disableConfirmWarning'
         />
     );
 
-    describe('initial state', () => {
-        test('with true', () => {
-            const props = {
-                ...baseProps,
-                value: true,
-            };
-
-            const wrapper = shallow(
-                <CustomEnableDisableGuestAccountsSetting {...props}/>,
-            );
+    describe('renders correctly', () => {
+        test('when enabled', () => {
+            const wrapper = renderComponent({value: true});
             expect(wrapper).toMatchSnapshot();
```

Comment:
What you want to match to the snapshot is `wrapper.container`, not the whole `wrapper`. The reality is that what `render` returns is not really a wrapper, but an object with several util functions. That is why the snapshot looks so weird.